### PR TITLE
Use statement cleanup

### PR DESCRIPTION
**Pull request by jrper**
As described in #87, this changeset reorders use statements in all modules to be in the Intel fortran compiler preferred order. This in turn gives slightly more overhead before running into internal compiler errors when additional code is added. A warn test has also been added to scan the ordering of the files as additional work is added.

### CODE REVIEW

**Comment by jrper** [comment]
Yes, I came to that realisation shortly after I managed to get the current version working. Do you consider that the maintainability benefits demand a rewrite?

**Comment by wence-** [comment]
Nah, probably not.

File: assemble/Adapt_State.F90 [hunk 1]
```diff
@@ -28,56 +28,53 @@
...
-  use data_structures
-  use detector_data_types
-  use detector_parallel
-  use diagnostic_variables
-  use intersection_finder_module
-  use diagnostic_variables
-  use diagnostic_fields_wrapper_new, only : calculate_diagnostic_variables_new => calculate_diagnostic_variables
-  use pickers
   use write_gmsh
 #ifdef HAVE_ZOLTAN
```
Comment (stephankramer) [inline assemble/Adapt_State.F90]: Yeah, so I was wondering about this: I presume you currently ignore `#ifdef` lines? So in this specific case the result of moving `mpi_interfaces` out of the `#ifdef HAVE_ZOLTAN` block is harmless - but how does this work generally?

Comment (jrper) [inline assemble/Adapt_State.F90]: Currently preprocessor protection is indeed ignored when the script calculates the necessary ordering. It's up to a human to put it in as necessary, and I missed it for `mpi_interfaces` here. That should be ok providing we never set up circular dependencies in the unpreprocessed module files, which I can't see a reason to do.

**Comment by jrper** [comment]
Included as part of #96.

**Pull request closed without merging**